ciao-controller/instance.go

```diff
@@ -359,8 +359,8 @@ func newConfig(ctl *controller, wl *types.Workload, instanceID string, tenantID
 		userData.Hostname = "cnci-" + tenantID
 	}
 
-	// handle storage resources in workload definition
-	if wl.Storage != nil {
+	// if no instance volumes specified, use those in workload
+	if len(volumes) == 0 && wl.Storage != nil {
```

Comment:
Good question.  Right now the workload doesn't include a boot_index.  So it's hard to reconcile with the cli's volume option that does include that.

Also I see the OpenStack API boot index argument as an option, especially when coupled with qemu/kvm, that is non-mandatory.  I'm not sure if/how we could really truly enforce boot ordering.  We can order the disks as we present them to qemu, but the actual VM disk enumeration and boot order is going to depend on qemu internals, the emulated bios, the VM's bootloader code and config, the kernel, disk labels, etc.

I don't think we have sufficiently specified supported use cases to declare an expected behavior here.  But:

The workload definition is our top level fundamental definition of the abstract thing that's going to run.  Where the abstract thing has an OS that is booted, IMHO that should be specified in the workload definition.  Were this not the case, it feels like it devolves to: does it make sense to have a workload defined as "Ubuntu 16.10 VM with 1vcpu and 16GB RAM", and then launch an instance of that workload with say a Fedora Cloud Base 24 iso attached, and hope for that iso to be the boot device.  I think no, but you could implement it.

It would help me more to understand the specific case you're trying to run:
* what's the workload definition in detail
* what's the volume list in the instance add cli invocation
* what's the expected outcome in your opinion for this context